### PR TITLE
Burn SelectedFileId in fire

### PR DESCRIPTION
**Pull request by izgzhen**
r? @Manishearth 

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

---

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/12898)

### CODE REVIEW

**Comment by izgzhen** [comment]
It's a bad title now I found ... I didn't burn it, I just aliased it. Should I _burn_ it?

**Comment by izgzhen** [comment]
In case it is not clear, I invent the wrapper before since at that time `Uuid` doesn't implement `Serialize/Deserialize`.

File: components/net_traits/filemanager_thread.rs [hunk 1]
```diff
@@ -98,10 +99,8 @@ impl RelativePos {
     }
 }
 
-// XXX: We should opt to Uuid once it implements `Deserialize` and `Serialize`
 /// FileID used in inter-process message
-#[derive(Clone, Debug, Deserialize, Serialize)]
-pub struct SelectedFileId(pub String);
+pub type SelectedFileId = Uuid;
```
Comment (Manishearth) [inline components/net_traits/filemanager_thread.rs]: Why keep the typedef at all?

**Comment by Manishearth** [comment]
Yes, I feel that it can be burned. Otherwise lgtm

**Comment by Manishearth** [comment]
Unit tests need updating

**Comment by izgzhen** [comment]
:/ `thread 'filemanager_thread::test_filemanager' panicked at 'Broken channel: Serde(Custom("bincode does not support Deserializer::deserialize"))', ../src/libcore/result.rs:788`

Thing is worse than I originally thought.

**Comment by izgzhen** [comment]
I am digging in `uuid` implementation. I will park this PR for a few days until I fixed this

**Comment by izgzhen** [comment]
bincode problem fixed. crates updated.

**Comment by izgzhen** [comment]
Tidy fixed.

**Comment by izgzhen** [comment]
@bors-servo retry

- https://github.com/servo/servo/issues/11574

**Comment by izgzhen** [comment]
@bors-servo retry

- https://github.com/servo/servo/issues/12891

**Comment by izgzhen** [comment]
@bors-servo retry

- again https://github.com/servo/servo/issues/12891 :(

**Comment by izgzhen** [comment]
Resolved

**Comment by izgzhen** [comment]
resolved

**Comment by Ms2ger** [comment]
Will reopen in a bit, homu got confused again

**Comment by izgzhen** [comment]
good to reopen

**Comment by Manishearth** [comment]
needs merge conflict fixes

**Comment by KiChjang** [comment]
Needs a new commit hash. @izgzhen can you do `git commit --amend -C HEAD` followed by `git push -f`?

**Comment by izgzhen** [comment]
I resolved the conflict and it looks like a new hash now to me

**Pull request merged**